src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -183,6 +122,153 @@ public GeneratorDriver AddAdditionalTexts(ImmutableArray<AdditionalText> additio
             return FromState(newState);
         }
 
+        public GeneratorDriverRunResult GetRunResult()
+        {
+            var trees = ArrayBuilder<SyntaxTree>.GetInstance();
+            var diagnostics = ArrayBuilder<Diagnostic>.GetInstance();
```

Comment:
Could specify ranges for these as well, it'll start them at a minimum of assuming that every generator will generate one tree? Might reduce resizing.

Reply:
We can't make that assumption. It's a pretty common case that there is nothing to generate.

---
In reply to: [466067485](https://github.com/dotnet/roslyn/pull/46296#discussion_r466067485) [](ancestors = 466067485)